src/main/java/org/sonatype/nexus/repository/r/internal/RHostedFacetImpl.java

```diff
@@ -75,35 +73,35 @@
   public Content getPackages(final String packagesPath) {
     checkNotNull(packagesPath);
     try {
+      // TODO: Do NOT do this on each request as there is at least some overhead, and memory usage is proportional to
+      // the number of packages contained in a particular path. We should be able to generate this when something has
+      // changed or via a scheduled task using an approach similar to the Yum implementation rather than this method.
```

Comment:
I think we do both. Taking the implementation from Yum saves us having to do any extra setup to ensure the metadata is generated but having the scheduled task is a useful fall-back if for some reason the metadata gets corrupted or Nexus gets interrupted while it is generating the metadata.